docs/book/misc/best-practices.md

```diff
@@ -58,52 +64,126 @@ if __name__ == "__main__":
 
 This ensures that loading the pipeline from elsewhere does not also run it.
 
-#### Don't use the same metadata stores across multiple artifact stores
+### Never call the pipeline instance `pipeline` or a step instance `step`
 
-You might run into issues as the metadata store will point to artifacts in
-inactive artifact stores.
+Doing this will overwrite the imported `pipeline` and `step` decorators and lead
+to failures at later stages if more steps and pipelines are decorated there.
+
+```python
+from zenml.pipelines import pipeline
+
+
+@pipeline
+def first_pipeline():
+    ...
+
+
+pipeline = first_pipeline(...)
+
+
+@pipeline  # The code will fail here
+def second_pipeline():
+    ...
+```
+
+### Explicitly set `enable_cache` at the `@pipeline` level
+
+Caching is enabled by default for ZenML Pipelines. It is good to be explicit
+though so that it is clear when looking at the code if caching is enabled or
+disabled for a given pipeline.
+
+### Explicitly disable caching when loading data from filesystem or external APIs
 
-#### Never call the pipeline instance `pipeline` or a step instance `step`
+ZenML inherently uses caching. However, this caching relies on changes of input
+artifacts to invalidate the cache. In case a step has external data sources like
+external APIs or filesystems, caching should be disabled explicitly for the 
+step.
 
-This will overwrite the imported `pipeline` and `step` decorators.
+### Enable cache explicitly for steps that have a `context` argument, if they don't invalidate the caching behavior
+
+Cache is implicitly disabled for steps that have a
+[context]((../developer-guide/step-fixtures.md#using-the-stepcontext)) argument,
+because it is assumed that you might use the step context to retrieve artifacts
+from the artifact store that are unrelated to the current step. However, if that
+is not the case, and your step logic doesn't invalidate the caching behavior, it
+would be better to explicitly enable cache for your step
+
+### Don't use the same metadata stores across multiple artifact stores
+
+You might run into issues as the metadata store will point to artifacts in
+inactive artifact stores.
 
-#### Use profiles to manage stacks
+### Use profiles to manage stacks
 
+Using profiles allows you to separate your zenml stack collections and work
```

Comment:
```suggestion
Using Profiles allows you to separate your ZenML stacks and work
```

Maybe don't introduce a new abstraction term that we don't in fact use anywhere else.